api/firmware/device.go

```diff
@@ -318,6 +318,33 @@ func (device *Device) Random() ([]byte, error) {
 	return randomResponse.RandomNumber.Number, nil
 }
 
+// RootFingerprint returns the keystore's root fingerprint, which is the first 32 bits of the
+// hash160 of the pubkey at the keypath m/.
+//
+// https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#key-identifiers
+func (device *Device) RootFingerprint() ([4]byte, error) {
```

Comment:
Did you consider using a slice instead? https://golang.org/doc/effective_go#arrays recommends using slices in most cases.

I get that you probably want to be specific about the return type but is a caller of `RootFingerprint` every going to index it with a `result[0]`? If the usage of the result is to always convert to a hex string or do some other slice manupulation, not indexing, I would use the slice type.

WDYT?

Reply:
It's just from my experience the code on the caller side looks uglier when dealing with arrays, compared to slices.

Reply:
Won't the caller just do `[:]`, e.g. `fingerprint[:]`, if they need a slice?

You are correct, I did it for better type safety.

Reply:
Yeah, you can always `[:]` but it just looks ugly.

> better type safety

but the `RootFingerprint` already guarantees in its doc comment it's always 32 bits. plus, the caller can and probably will almost do `[:]` anyway, so arguably not much more type safety than with a slice. The arrays are used mostly for a very specific memory/struct layouts, not type safety.

I don't might either if you really prefer an array here.

Reply:
Another slightly ugly thing is you always have to do

```go
if err != nil {
  return result, err
}
```

instead of

```go
if err != nil {
  return nil, err
}
```

the second version obvously reads nicer and more idiomatic.

Reply:
done

Reply:
:pray: